### DISCUSSION TITLE
Trying to extract data from a PDF file

### DISCUSSION THREAD

**Discussion by jakobdo** [jsvine/pdfplumber#625] [Ask for help with specific PDFs]
Hello, I have been using this library before and I am really amazed how "easy" it is to extract data. But!
When data is not easily extracted, I find it hard to tweak these settings and get the data I need.
Instead of getting a working solution, how will I be able to extract the data from this pdf: https://www.taggmbh.at/fileadmin/content/TAG-Website-Content-SM/2022_Maintenance_PROD_PDF.pdf

When using the debug-table-finder, the last row on page 1 is missing:
![2022_Maintenance_PROD_PDF_1](https://user-images.githubusercontent.com/278270/158386488-69e6c3cd-dddf-4eb4-9dd0-07f9b75a4f5d.png)

How do I tweak the table settings to get the last row/line?
I have read about using explicit_lines, but I need to get the lines for a start.

**Comment by jsvine** [comment]
Hi @jakobdo, and glad to hear that you've found the library useful! For your particular example, I'd recommend getting the position of the PDF's missing final line this way, by identifying the bottom-most extremity of existing `page.rects` objects (since those lines are represented as `rect` objects):

```python
line_pos = max(r["bottom"] for r in page.rects)
table = page.extract_table({
    "explicit_horizontal_lines": [ line_pos ]
})
```

Demonstrating via `.debug_tablefinder(...)`:

```python
im = page.to_image()
line_pos = max(r["bottom"] for r in page.rects)
im.reset().debug_tablefinder({
    "explicit_horizontal_lines": [ line_pos ]
})
```

![image](https://user-images.githubusercontent.com/534702/158482961-c4d00528-8190-4419-9c01-b566fa3fd220.png)

**Reply by jakobdo** [reply]
@jsvine works like a charm like always. I might create another question in a few seconds, because I think I have "another" PDF which is producing some weird values. (characters are double)
